esmf_regrid/_grid.py

```diff
@@ -0,0 +1,179 @@
+# -*- coding: utf-8 -*-
+
+import cartopy.crs as ccrs
+import ESMF
+from iris.exceptions import CellMeasureNotFoundError
+import numpy as np
+
+
+def _bounds_cf_to_simple_1d(cf_bounds):
+    assert (cf_bounds[1:, 0] == cf_bounds[:-1, 1]).all()
+    simple_bounds = np.empty((cf_bounds.shape[0]+1,), dtype=np.float64)
+    simple_bounds[:-1] = cf_bounds[:, 0]
+    simple_bounds[-1] = cf_bounds[-1, 1]
```

Comment:
Compare the iris.coords.Coord methods [is_contiguous](https://github.com/SciTools/iris/blob/077df3ec15530255df80c2fc330eb21998e24769/lib/iris/coords.py#L1732) and [contiguous_bounds](https://github.com/SciTools/iris/blob/077df3ec15530255df80c2fc330eb21998e24769/lib/iris/coords.py#L1762). Notably, the Iris version allows for a degree of tolerance. It might be worth incorporating the Iris versions for consistencies sake.